### PR TITLE
[frontend] Make it possible to configure visibility of the help menu …

### PR DESCRIPTION
**Pull request by tarunjangid**
…in the sidebar

## What changes were proposed in this pull request?

- There was a request in our community to have a feature to remove the help tip - https://github.com/cloudera/hue/issues/3647
- Currently, the "Help" menu in the sidebar is visible to all the users.
- In this PR, a new configuration named **is_help_menu_enabled** is added to control the help menu.
- If this configuration is set to **False**, we cannot see the Help menu as below:
Screenshot 2024-04-30 at 1 18 19 PM

- If the configuration is set to **True** (the default value), the Help menu is visible as usual:
Screenshot 2024-04-30 at 1 20 11 PM

## How was this patch tested?

- manually tested in my local setup

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

File: desktop/conf.dist/hue.ini [hunk 1]
```diff
@@ -334,6 +334,10 @@ http_500_debug_mode=false
 # default value is false
 ## disable_local_storage = false
 
+# Whether or not to show the Help menu in the Sidebar.
+# Default value is true
+## is_help_menu_enabled=true
```
Comment (Harshg999) [inline desktop/conf.dist/hue.ini:339]: Nit: Can we change it to `enable_help_menu=true`?

File: desktop/conf/pseudo-distributed.ini.tmpl [hunk 2]
```diff
@@ -339,6 +339,10 @@
   # default value is false
   ## disable_local_storage = false
 
+  # Whether or not to show the Help menu in the Sidebar.
+  # Default value is true
+  ## is_help_menu_enabled=true
```
Comment (Harshg999) [inline desktop/conf/pseudo-distributed.ini.tmpl:344]: Same here `enable_help_menu`

File: desktop/core/src/desktop/conf.py [hunk 3]
```diff
@@ -2156,6 +2156,13 @@ def default_gist_preview():
          "please make this value true in case local storage should not be used.")
 )
 
+IS_HELP_MENU_ENABLED = Config(
+  key='is_help_menu_enabled',
```
Comment (Harshg999) [inline desktop/core/src/desktop/conf.py:2160]: Same here `ENABLE_HELP_MENU`

File: desktop/core/src/desktop/templates/global_js_constants.mako [hunk 4]
```diff
@@ -798,6 +798,7 @@
   window.IS_LDAP_SETUP = '${ 'desktop.auth.backend.LdapBackend' in conf.AUTH.BACKEND.get() }' === 'True';
   window.LOGGED_USERNAME = '${ user.username }';
   window.LOGGED_USER_ID = ${ user.id };
+  window.IS_HELP_MENU_ENABLED = '${conf.IS_HELP_MENU_ENABLED.get()}' === 'True';
```
Comment (Harshg999) [inline desktop/core/src/desktop/templates/global_js_constants.mako:801]: `window.ENABLE_HELP_MENU`

**Review by Harshg999 — COMMENTED** [review]
The config part looks good to me, try adding unit tests for UI part? Also please fix the failing CI checks.

@bjornalm @JohanAhlen Can you review the UI part? Thanks.

**Review by bjornalm — APPROVED** [review]
Nice work!

**Pull request merged**